lib/contextual_logger/redactor.rb

```diff
@@ -17,12 +19,30 @@ def register_secret(sensitive_data)
       end
     end
 
-    def redact(log_line)
+    def redact(log_entry)
       if redaction_regex
-        log_line.gsub(redaction_regex, '<redacted>')
+        case log_entry
+        when Hash
+          log_entry.reduce({}) do |redacted_log_entry, (key, value)|
+            redacted_log_entry[key] = redact(value)
+            redacted_log_entry
+          end
+        when Array
+          log_entry.map { |value| redact(value) }
+        when true, false
```

Comment:
Hmm, are you sure we need to support true/false? But not integers or floats or nil?